backend/src/controllers/auth-controller.ts

```diff
@@ -62,4 +66,80 @@ export const loginUser = async (req: Request, res: Response) => {
     } catch (error: any) {
         res.status(500).json({ message: 'Server error', error: error.message });
     }
-}
\ No newline at end of file
+}
+
+export const postReset = async (req: Request, res: Response) => {
+    try {
+        crypto.randomBytes(32, (err, buf) => {
+           if (err){
+               console.log(err);
+           }
+           const token = buf.toString('hex');
+           // @ts-ignore
+            User.findOne({email: req.body.email}).then(user => {
```

Comment:
Don't use `.then()` and `.catch()` syntax in `async` JS. Use `async/await` instead